firefox-ios/Client/Ecosia/Extensions/BrowserViewController+Ecosia.swift

```diff
@@ -237,4 +243,48 @@ extension BrowserViewController {
             }
             .logout()
     }
+
+    private func handleProfilePageDetection(_ url: URL) {
+        EcosiaLogger.auth.info("🔐 [WEB-PROFILE] Profile URL detected in navigation: \(url)")
+        EcosiaLogger.auth.info("🔐 [WEB-PROFILE] Opening native profile modal")
+
+        DispatchQueue.main.async { [weak self] in
+            self?.presentProfileModal()
+        }
+    }
+}
+
+// MARK: Profile Modal Presentation
+extension BrowserViewController {
+    /// Presents the profile page as a modal, similar to EcosiaAccountImpactView
+    func presentProfileModal() {
+        guard #available(iOS 16.0, *) else {
+            EcosiaLogger.auth.notice("Profile modal requires iOS 16.0+")
+            return
+        }
```

Comment:
Should we fallback to handling it via web here? Or do we just don't care about iOS 15 for this case?